examples/morphing_basis/morphing_example.ipynb

```diff
@@ -0,0 +1,373 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Morphing is a functionality implemented by madminer that allows for the fast and exact interpolation of event weights (or any derived quantity) from a set of original benchmark points (the “morphing basis”) to anywhere in theory parameter space. In this jupyter notebook, we will be focus on how to utilizing the morphing function of madminer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Preparation\n",
+    "import numpy as np\n",
+    "\n",
+    "from madminer.utils import morphing as m"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The current morphing function can take in two different format of inputs. One format is setting basis points default as work for both and production and decay, with the basis points input shape of (n_basis_points, n_parameters); The other allow user to specify couplings groups of production (gp), decay(gd) and same (gs, work as both production and decay) separately, each with a shape of (n_parameters, n_basis_points). Both ways support overdetermine morphing, which allows user to input as many basis_points as possible for better results. Both methods should output the same matrix and weights values. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First Input Format"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Example of using the first input format that default couplings that work for both decay and production couplings.\n",
+    "\n",
+    "# First, we should create a new morpher object using the code below\n",
+    "morpher_1 = m.PhysicsMorpher(parameter_max_power=[2, 2])\n",
+    "\n",
+    "# Components is the power of each parameter in the given coupling combination, for example, if the equation is gs_0^2 * gs_1^3, the components for this combination is [2,3]\n",
```

Comment:
Please try to keep the line lengths of these comments a bit shorter to avoid rendering issues.

Reply:
thank you, I had reduced the lengths of many comments.